stdlib/system/src/test/java/org/ballerinalang/stdlib/system/FileSystemTest.java

```diff
@@ -0,0 +1,382 @@
+        BValue[] args = {new BString(filepath.toString())};
+        BValue[] returns = BRunUtil.invoke(compileResult, "testCreateFile", args);
+        assertTrue(returns[0] instanceof BError);
+        BError error = (BError) returns[0];
+        assertEquals(error.getReason(), "{ballerina/system}INVALID_OPERATION");
+        log.info("Ballerina error: " + error.getDetails().stringValue());
+    }
+
+    @Test(description = "Test for creating new file")
+    public void testCreateDirWithParentDir() throws IOException {
+        Path filepath = tempDirPath.resolve("temp-dir").resolve("nested-dir");
+        FileUtils.deleteDirectory(filepath.toFile());
+        try {
+            BValue[] args = {new BString(filepath.toString()), new BBoolean(true)};
+            BRunUtil.invoke(compileResult, "testCreateDir", args);
+            assertTrue(Files.exists(filepath));
+        } finally {
+            FileUtils.deleteDirectory(filepath.getParent().toFile());
+        }
+    }
+
+    @Test(description = "Test for creating new file")
+    public void testCreateDirWithoutParentDir() throws IOException {
+        Path filepath = tempDirPath.resolve("temp-dir").resolve("nested-dir");
+        FileUtils.deleteDirectory(filepath.getParent().toFile());
+        try {
+            BValue[] args = {new BString(filepath.toString()), new BBoolean(false)};
+            BValue[] returns = BRunUtil.invoke(compileResult, "testCreateDir", args);
+            assertTrue(returns[0] instanceof BError);
+            BError error = (BError) returns[0];
+            assertEquals(error.getReason(), "{ballerina/system}FILE_SYSTEM_ERROR");
+            log.info("Ballerina error: " + error.getDetails().stringValue());
+            assertFalse(Files.exists(filepath));
+        } finally {
+            FileUtils.deleteDirectory(filepath.getParent().toFile());
+        }
+    }
+
+    @Test(description = "Test for copying a file to new location")
+    public void testCopyFile() throws IOException, InterruptedException {
+        Path tempDestPath = null;
+        try {
+            Path sourcePath = Paths.get("src", "test", "resources", "data-files", "src-file.txt");
+            tempDestPath = tempDirPath.resolve("dest-file.txt");
+
+            BValue[] args = {new BString(sourcePath.toString()), new BString(tempDestPath.toString()),
+                    new BBoolean(false)};
+            BRunUtil.invoke(compileResult, "testCopy", args);
+            assertTrue(Files.exists(tempDestPath));
+            assertTrue(Files.exists(sourcePath));
+            assertEquals(tempDestPath.toFile().length(), sourcePath.toFile().length());
+
+            // Execute same with replaceExist false
+            long modifiedTime1 = tempDestPath.toFile().lastModified();
+            BRunUtil.invoke(compileResult, "testCopy", args);
+            long modifiedTime2 = tempDestPath.toFile().lastModified();
+            assertEquals(modifiedTime2, modifiedTime1);
```

Comment:
Shall we rename variables as `modifedTime` and `modifiedTimeAfterCopy` so that it is more clear instead of using numbers as 1 and 2. Also in other places?